yang/frr-ospfd.yang

```diff
@@ -0,0 +1,970 @@
+              "Shortcut ABR.";
+          }
+          enum "standard" {
+            value 4;
+            description
+              "Standard behavior (RFC2328).";
+          }
+        }
+      }
+
+      leaf opaque-lsa {
+        type boolean;
+        description
+          "Enable the Opaque-LSA capability (RFC2370)";
+      }
+
+      leaf rfc1583compatibility {
+        type boolean;
+        description
+          "Enable the RFC 1583 compatibility flag.";
+      }
+
+      leaf router-id {
+        type inet:ipv4-address;
+        description
+          "Router-id for the OSPF process.";
+      }
+
+      leaf write-multiplier {
+        type uint8 {
+          range "1..100";
+        }
+        description
+          "Write multiplier.";
+      }
+    }
+  }
+
+  grouping timer-group {
+    container timers {
+      leaf refresh-interval {
+        type uint16 {
+          range "10..1800";
+        }
+        units "seconds";
+        description
+          "The maximum time between distinct originations of any particular
+           LSA, value in units seconds.";
+      }
+
+      leaf lsa-min-arrival {
+        type uint32 {
+          range "0..600000";
+        }
+        units "milliseconds";
+        description
+          "Minimum delay in receiving new version of an LSA.";
+      }
+
+      container throttle {
+        leaf lsa-all {
+          type uint16 {
+            range "0..5000";
+          }
+          units "milliseconds";
+          description
+            "LSA delay between transmissions.";
+        }
+
+        leaf spf {
+          type uint32 {
+            range "0..600000";
+          }
+          units "milliseconds";
+          description
+            "Delay from first change received till SPF calculation.";
+        }
+      }
+    }
+  }
+
+  grouping segment-routing-group {
+    container segment-routing {
+      container global-block {
+        description
+          "Segment Routing Global Block label range.";
+        must "./upper-bound > ./lower-bound";
+        leaf lower-bound {
+          type uint32 {
+            range "0..1048575";
+          }
+        }
+
+        leaf upper-bound {
+          type uint32 {
+            range "0..1048575";
+          }
+        }
+      }
+
+      leaf node-msd {
+        type uint8 {
+          range "1..16";
+        }
+        description
+          "Maximum Stack Depth for this router.";
+      }
+
+      leaf on {
+        type boolean;
+        description
+          "Enable Segment Routing.";
+      }
+
+      list prefix-sid {
+        key "index";
+        leaf index {
+          type uint16;
+          description
+            "SID index for this prefix.";
+        }
+
+        leaf prefix {
+          type inet:ipv4-prefix;
+          description
+            "Prefix SID.";
+        }
+
+        leaf no-php-flag {
+          type boolean;
+          description
+            "Don't request Penultimate Hop Popping (PHP).";
+        }
```

Comment:
PR #6726 adds support for explicit-null flag. Can you change the leaf by an enumerate like in IS-IS Segment Routing yang model ?